portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,22 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Reduces opacity and shows name of location when image is moused over.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function displayLocation(name) {
+  var label = document.getElementById(name); 
+  var img = document.getElementById(name+'-Img'); 
+  label.style.display = "block"; 
+  img.style.opacity = 0.3; 
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/**
+ * Hides label and changes opacity back to 1 when mousing out of image. 
+ */
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+function hideLocation(name) {
+  var label = document.getElementById(name); 
+  var img = document.getElementById(name+'-Img'); 
```

Comment:
I noticed that your hideLocation and displayLocation functions both use this logic. It might be helpful to create a getLocation private function that both of these can use.

Reply:
Just added!

Reply:
Ah i didn't realize they were getting slightly different elements. Nice job returning them in an array structure and destructuring at the call site!